crates/derive/src/online/mod.rs

```diff
@@ -3,11 +3,42 @@
 /// Prelude for online providers.
 pub(crate) mod prelude {
     pub use super::{
-        AlloyChainProvider, AlloyL2ChainProvider, BeaconClient, OnlineBeaconClient,
-        OnlineBlobProvider, SimpleSlotDerivation,
+        new_online_stack, AlloyChainProvider, AlloyL2ChainProvider, BeaconClient,
+        OnlineBeaconClient, OnlineBlobProvider, SimpleSlotDerivation,
     };
 }
 
+use crate::{
+    stages::{
+        AttributesBuilder, AttributesQueue, BatchQueue, ChannelBank, ChannelReader, FrameQueue,
+        L1Retrieval, L1Traversal, NextAttributes,
+    },
+    traits::{
+        ChainProvider, DataAvailabilityProvider, L2ChainProvider, OriginProvider, ResettableStage,
+    },
+    types::RollupConfig,
+};
+use alloc::sync::Arc;
+use core::fmt::Debug;
+
+/// Creates a new [OnlineStageStack].
+#[cfg(feature = "online")]
+pub fn new_online_stack(
+    rollup_config: Arc<RollupConfig>,
+    chain_provider: impl ChainProvider + Clone + Debug + Send,
+    dap_source: impl DataAvailabilityProvider + OriginProvider + Clone + Debug + Send,
+    fetcher: impl L2ChainProvider + Clone + Debug + Send,
+    builder: impl AttributesBuilder + Clone + Debug + Send,
```

Comment:
Can we provide the online implementations of these for the caller? i.e. an online stack uses the `AlloyChainProvider`, etc.

Reply:
Great callout - an oversight on my part since I adapted generic code to be online 😄

Reply:
Added concrete types for all except the dap source since we don't have an alloy-backed dap impl. Will stack a pr ontop to add that.